worker/sshserver/server.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	"github.com/gliderlabs/ssh"
+	"github.com/juju/errors"
+	gossh "golang.org/x/crypto/ssh"
+	"gopkg.in/tomb.v2"
+)
+
+// ServerWorkerConfig holds the configuration required by the server worker.
+type ServerWorkerConfig struct {
+	Logger Logger
+}
+
+// Validate validates the workers configuration is as expected.
+func (c ServerWorkerConfig) Validate() error {
+	if c.Logger == nil {
+		return errors.NotValidf("Logger is required")
+	}
+	return nil
+}
+
+// ServerWorker is a worker that runs an ssh server.
+type ServerWorker struct {
+	tomb tomb.Tomb
+
+	// Server holds the embedded server.
+	Server *ssh.Server
+
+	// config holds the configuration required by the server worker.
+	config ServerWorkerConfig
+}
+
+// NewServerWorker returns a running embedded SSH server.
+func NewServerWorker(config ServerWorkerConfig, startServer bool) (*ServerWorker, error) {
+	if err := config.Validate(); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	s := &ServerWorker{config: config}
+	s.Server = &ssh.Server{
+		Addr: ":2223",
```

Comment:
should we not default to port 17022?

Reply:
It doesn't matter for now, I'll plug the config in once the config PR is merged.